processor/zabbix.go

```diff
@@ -0,0 +1,180 @@
+package processor
+
+import (
+	"errors"
+	"github.com/buger/jsonparser"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/devopsext/events/common"
+	sreCommon "github.com/devopsext/sre/common"
+)
+
+type ZabbixProcessor struct {
+	outputs  *common.Outputs
+	tracer   sreCommon.Tracer
+	logger   sreCommon.Logger
+	requests sreCommon.Counter
+	errors   sreCommon.Counter
+}
+
+type zabbixEvent struct {
+	EventType          string
+	EventNSeverity     string
+	Status             string
+	EventID            string
+	HostName           string
+	ItemID             string
+	ItemLastValue      string
+	AlertURL           string
+	TriggerDescription string
+	EventTags          string
+	Scope              string
+}
+
+func ZabbixProcessorType() string {
+	return "Zabbix"
+}
+
+func (p *ZabbixProcessor) EventType() string {
+	return common.AsEventType(ZabbixProcessorType())
+}
+
+func (p *ZabbixProcessor) send(span sreCommon.TracerSpan, channel string, o interface{}, t *time.Time) {
+
+	e := &common.Event{
+		Channel: channel,
+		Type:    p.EventType(),
+		Data:    o,
+	}
+	if t != nil && (*t).UnixNano() > 0 {
+		e.SetTime((*t).UTC())
+	} else {
+		e.SetTime(time.Now().UTC())
+	}
+	if span != nil {
+		e.SetSpanContext(span.GetContext())
+		e.SetLogger(p.logger)
+	}
+	p.outputs.Send(e)
+}
+
+func (p *ZabbixProcessor) HandleEvent(e *common.Event) error {
+
+	if e == nil {
+		p.logger.Debug("Event is not defined")
+		return nil
+	}
+	p.requests.Inc(e.Channel)
+	p.outputs.Send(e)
+	return nil
+}
+
+func (p *ZabbixProcessor) HandleHttpRequest(w http.ResponseWriter, r *http.Request) error {
+
+	span := p.tracer.StartChildSpan(r.Header)
+	defer span.Finish()
+
+	channel := strings.TrimLeft(r.URL.Path, "/")
+	p.requests.Inc(channel)
+
+	var body []byte
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+
+	if len(body) == 0 {
+		p.errors.Inc(channel)
+		err := errors.New("empty body")
+		p.logger.SpanError(span, err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return err
+	}
+
+	p.logger.SpanDebug(span, "Body => %s", body)
+
+	var request zabbixEvent
```

Comment:
we should use ZabbixEvent with capital firts letter

Reply:
fix it